### PR TITLE
Ability to suspend maid service

### PR DESCRIPTION
**Pull request by Ada18980**
This change adds an extra dialogue option in the Maid Quarters, allowing you to prevent the maids from helping you for a number of time periods:
- 10 minutes
- 30 minutes
- 1 hour

The effects of disabling maid service are:
- No help from the hallway maid (she will refuse to release you and tell you how much time is remaining)
- No help from photography club or the maid quarters maid 
- No chastity removal from the management

There is a full dialogue tree (english only, currently) with several chances for the player to back out before deciding to disable maids, and there are no options that are indefinite; the longest is 1 week. 

When talking to the maid, the player can also "Shake their head and plead for more" which allows them to refresh the timer for the maids being disabled. This is helpful for players who don't wish to remove their gags in order to talk to the maid in the maid quarters (or can't)

Test cases:
- Tested asking for maid work while both gagged and not gagged
- Tested photography club (refuses to help)
- Tested asking for chastity removal in Management (refuses to help)
- Tested being both gagged and ungagged when being helped by maid
- Tested owner locks and lover locks with maid (it should override it and prevent the player from removing owner locks while maid help is disabled), gagged and ungagged
- Tested being able to remove chastity after the maid status was removed
- Tested the maid status expiring after 10 minutes
- Tested being able to cancel owners and lovers while the maids are disabled (although not when gagged).

Known issues:
- Being gagged with an owner lock will softlock you unless your owner helps you. It might be possible to get stuck for a week in the worst case scenario, but disabling the maids for a week is entirely the user's own choice.
- A week seems like it might be too dangerous and may get a lot of console help requests. The three days option might make more sense to cap it. Alternatively, disabling the maids for a week could be made more difficult to reduce the incidence of users choosing the option unwisely.
- Players might get softlocked for a week if their owner is never online and they chose to be helpless for a week, especially if wearing an owner locked gag
- Combination locks may be dangerous for users without owners, as there will be no easy way to remove them without waiting it out.
- There is no way currently for the player to disable maids while gagged, aside from extending it by talking to the hallway maid and nodding.
- Random kidnappers will remove items with lover's locks. This isn't a bug per se but may reduce enjoyment for people who prefer to be trapped in lover's items. This issue is outside the scope of this PR but is worth noting.

### CODE REVIEW

**Comment by Ada18980** [comment]
Reduced to 1 hour max and gambling hall no longer frees the player.

File: BondageClub/Screens/Room/Gambling/Gambling.js [hunk 1]
```diff
@@ -20,6 +20,14 @@ var GamblingAppearancePlayer = null;
 var GamblingIllegalChange = false; // Sub Player lost Cloth although forbidden by Mistress
 var GamblingToothpickCount = 0; // available Toothpicks
 
+// Returns TRUE if the player has maids disabled
+/**
+ * Checks if the player is helpless (maids disabled) or not.
+ * @returns {boolean} - Returns true if the player still has time remaining after asking the maids to stop helping in the maid quarters
+ */
+function GamblingIsMaidsDisabled() {  var expire = LogValue("MaidsDisabled", "Maid") - CurrentTime ; return (expire > 0 || expire > 604800000)  }
```
Comment (Nina-1474) [inline BondageClub/Screens/Room/Gambling/Gambling.js:28]: Doesn't the > 0 here make the > 604800000 redundant?

File: BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv [hunk 2]
```diff
@@ -44,19 +44,23 @@ ClubSlave,,(Leave her.),,DialogLeave(),,
...
+0,,(Nod yes.),(She chuckles) Not so fast girl. We can't help you until your time is up.,,CanUngagAndMaidsDisabled(),
 0,,"Hello, I'm DialogPlayerName.  Pleased to meet you.","Pleased to meet you too.  I'm DialogCharacterName, I'm responsible to find work for maids.",DialogRemove(),"DialogReputationLess(""Maid"", 9)",
 0,,What's going on in here?,Maids come here to get job assignments.  They can earn some money by working for the club.,,,
 0,,What kind of chores do maid do in the club?,"Cleaning, serving drinks, testing new devices, rescuing helpless people and more.",,,
 0,10,Can I become a maid?,"You could, if you're not afraid of work and you don't mind the uniform.",,"DialogReputationLess(""Maid"", 0)",
-0,100,Do you have work for me?,"Absolutely, these are the chores available for you.  (She shows you a list of maid tasks to complete.)",,"DialogReputationGreater(""Maid"", 1)",
+0,100,Do you have work for me?,"Absolutely, these are the chores available for you.  (She shows you a list of maid tasks to complete.)",,CanDoWorkForMaids()
+0,0,Do you have work for me?,"No, you can't work for the maids if you are on the no-help list",,MaidQuartersCanDoWorkButMaidsDisabled()
```
Comment (Nina-1474) [inline BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv:54]: Should the MaidQuarters part be in the function name here? There are also other cases in this PR where the room name is appearing in the csv file function references.

File: BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv [hunk 3]
```diff
@@ -247,3 +251,35 @@ ClubSlave,,(Leave her.),,DialogLeave(),,
...
+604,607,So there's a catch?,"Yes. We will add your name to our no-help list. You can set the amount of time it will be active for up to one week.",,,
+607,610,Let me try this for 10 minutes and see how it feels,"You will be helpless for ten minutes, and the maids will not rescue you, nor will our associates in the photography club assist you. Is this what you want?",,,
+607,611,Give me half an hour,"You will be helpless for 30 minutes, and the maids will not rescue you, nor will our associates in the photography club assist you. Is this what you want?",,,
+607,612,I want to be bound for one hour,"You will be helpless for 1 hour, and the maids will not rescue you, nor will our associates in the photography club assist you. Is this what you want?",,,
+607,607,I was looking for something more... permanent.,"(She giggles) Legally we can only keep you so long.",,,
+607,0,This seems too much for me. (Go back),"(She nods) Everyone is different. There's no shame in that.",,,
+
+610,607,Let me consider the other options first...,"We can add your name to our no-help list. You can set the amount of time it will be active for up to one week.",,,
```
Comment (Nina-1474) [inline BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv:268]: Is the plan to eventually extend the limit beyond an hour? There's a lot of dialog and code still referencing more than that, so unless we plan to use them in future they should probably be removed.

File: BondageClub/Screens/Room/Gambling/Gambling.js (hunk 1 above)
Comment (Ada18980) [reply BondageClub/Screens/Room/Gambling/Gambling.js:28]: Yes, it does. Removing~

File: BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv (hunk 3 above)
Comment (Ada18980) [reply BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv:268]: Yes, the plan is to be able to increase it in the future if there is significant demand for it.

File: BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv (hunk 2 above)
Comment (Ada18980) [reply BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv:54]: As far as I know it works when I tested it. In the Photographic dialog file there are many functions which have Photographic in the name. I am not sure if it breaks it or not

Comment (Nina-1474) [reply BondageClub/Screens/Room/MaidQuarters/Dialog_NPC_MaidQuarters_Maid.csv:54]: On closer inspection, leaving the room name in these functions is actually ok too.

**Comment by Ben987** [comment]
I'm curious to try this one, lets hope it won't lead to players getting stuck.  I will test it soon, thanks!

**Pull request merged**